src/scheduler/fifo.c

```diff
@@ -2355,6 +2351,46 @@ next_job(status *policy, server_info *sinfo, int flag)
 	return rjob;
 }
 
+/**
+ * @brief
+ *	Helper function used to copy the given source string to the destination string. It also
+ *	frees the destination and then allocates required memory before copying.
+ *
+ * @param[in] dest - Address of pointer to destination string
+ * @param[in] dest - pointer to source string
+ *
+ * @retval
+ * @return 0 - Failure
+ * @return  1 - Success
+ *
+ * @par Side Effects:
+ *	None
+ *
+ *
+ */
+static int
+copy_attr_value(char **dest, char *src)
+{
+	int len = 0;
+	int ret = 0;
+
+	if (*dest != NULL)
+		free(*dest);
+
+	if (src !=NULL) {
+		len = strlen(src);
+		*dest = (char*)malloc(len + 1);
+		if (*dest == NULL) {
+			log_err(errno, __func__, MEM_ERR_MSG);
```

Comment:
Don't use log_err(), use schdlog().

Reply:
Most of the places where the macro MEM_ERR_MSG is used, it seems we are using log_err() only.  I am fine to do the change but eager to know the exact difference between these two in terms of usage. As far as I know schdlog() logs to both the scheduler logs as well as console. Is this something that we actually want in this case?

Reply:
The scheduler doesn't use log_events like the server does.  It has its own thing (log_filter).  The schdlog() function filters based on it.  If you use log_err(), then there is no way to filter these messages if it is desired.

Reply:
Thanks for letting me know this. I have made the changes accordingly. It seems there are several places where still log_err() is used with MEM_ERR_MSG which lead to my initial confusion.